src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -187,7 +187,13 @@ public FormValidation doCheckUrl(@AncestorInPath Item item,
                     .getClient();
             StandardCredentials credential = lookupCredentials(item, credentialsId, url);
             git.addDefaultCredentials(credential);
-            CredentialsProvider.track(item, credential);
+
+            // Should not track credentials use in any checkURL method, rather should track
+            // credentials use at the point where the credential is used to perform an
+            // action (like poll the repository, clone the repository, publish a change
+            // to the repository).
+            //
+            // CredentialsProvider.track(item, credential);
```

Comment:
I'd remove this line. The comment should suffice to prevent others trying to add back